tests/test_basic_ops_dict.py

```diff
@@ -287,6 +300,60 @@ def test_items_with_interpolation() -> None:
     assert r["b"] == 2
 
 
+@mark.parametrize(
+    ("cfg", "expected", "expected_no_resolve"),
+    [
+        param({}, [], [], id="empty"),
+        param({"a": 10}, [("a", 10)], [("a", 10)], id="simple"),
+        param(
+            {"a": 2, "b": "${a}"},
+            [("a", 2), ("b", 2)],
+            [("a", 2), ("b", "${a}")],
+            id="interpolation_in_value",
+        ),
+        param(
+            {"a": "???"},
+            raises(MissingMandatoryValue),
+            [("a", "???")],
+            id="missing_value",
+        ),
+        # Special DictConfigs
+        param(DictConfig(None), raises(TypeError), raises(TypeError), id="none"),
+        param(
+            DictConfig("???"),
+            raises(MissingMandatoryValue),
+            raises(MissingMandatoryValue),
+            id="missing",
+        ),
+        param(DictConfig("${missing}"), [], [], id="missing_interpolation"),
+        param(
+            DictConfig("${a}", parent=DictConfig({"a": {"b": 10}})),
+            [],
+            [],
+            id="missing_interpolation",
+        ),
+    ],
+)
+def test_items(cfg: Any, expected: Any, expected_no_resolve: Any) -> None:
+    cfg = _ensure_container(cfg)
+
+    if isinstance(expected, list):
+        assert list(cfg.items()) == expected
+    else:
+        with expected:
```

Comment:
What happened to "better handle error situations in a different test"? :P
(not asking you to change it, but curious what is the reasoning you apply when deciding to split or not)

Reply:
Don't be annoying :P.

> not asking you to change it, but curious what is the reasoning you apply when deciding to split or not

In this case, the test is testing one input against two outputs. one or both of which can fail.
The overall complexity of the test code is still low.
I think splitting into error cases would result in 3 times more test code without any meaningful gain in test simplicity.